content/client-lib-development-guide/features.textile

```diff
@@ -682,18 +693,17 @@ h4. ChannelOptions
 * @(TB1)@ options provided when instancing a channel, see "Java ChannelOptions":https://github.com/ably/ably-java/blob/master/src/io/ably/types/ChannelOptions.java as a reference
 * @(TB2)@ The attributes of @ChannelOptions@ consist of:
 ** @(TB2a)@ @encrypted@ boolean - when true, enables automatic encryption & decryption of all messages
-** @(TB2b)@ @cipherParams@ @CipherParams@ - when encrypted is true, the cipher params are required
+** @(TB2b)@ @cipherParams@ - This is required if @encrypted@ is true. It consists of either:
+*** @(TB2b1)@ A @CipherParams@ instance, or
+*** @(TB2b2)@ an options hash (or language equivalent) consisting of the key @key@ with value a key. In this case, the client library should call `getDefaultParams`, passing it the key, to obtain a @CipherParams@ instance
```

Comment:
OK. Attempting to synthesize all of the above comments:

- getDefaultParams must be able to take an arbitrary subset of CipherParams fields as a hashmap or equivalent that includes a key; will validate that the provided options are valid and self-consistent as best it can, and will return a complete CipherParams instance.
- if requiring a hash map is unidiomatic, getDefaultParams may optionally also take a key only.
- The `cipher` field of channelOptions must consist of either a CipherParams object, or an arbitrary subset of CipherParams fields that includes a key. In the latter case, the channel will pass it through getDefaultParams to generate a CipherParams.
- if requiring a hash map is unidiomatic, the client lib may optionally provide an alternative constructor for channelOptions that takes a key only. If it does this, it must be differentiated from the normal constructor such that it is clear that the value being passed in is a key.

Everyone's happy with that? If so I'll add it to the PR